assume/scenario/loader_pypsa.py

```diff
@@ -0,0 +1,232 @@
+# SPDX-FileCopyrightText: ASSUME Developers
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import shutil
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import pandas as pd
+import pypsa
+from dateutil import rrule as rr
+
+from assume import World
+from assume.common.forecasts import NaiveForecast
+from assume.common.market_objects import MarketConfig, MarketProduct
+
+
+async def load_pypsa_async(
+    world: World,
+    scenario: str,
+    study_case: str,
+    network: pypsa.Network,
+    marketdesign: list[MarketConfig],
+):
+    """
+    This initializes a scenario from the given pypsa grid.
+
+    Args:
+        world (World): the world to add this scenario to
+        scenario (str): scenario name
+        study_case (str): study case name
+        network (str): database uri to connect to the OEDS
+        marketdesign (list[MarketConfig]): description of the market design which will be used with the scenario
+    """
+    index = network.snapshots
+    index.freq = "h"
+    start = index[0]
+    end = index[-1]
+    sim_id = f"{scenario}_{study_case}"
+    print(f"loading scenario {sim_id}")
+
+    await world.setup(
+        start=start,
+        end=end,
+        save_frequency_hours=4,
+        simulation_id=sim_id,
+        index=index,
+    )
+    # setup eom market
+
+    mo_id = "market_operator"
+    world.add_market_operator(id=mo_id)
+    grid_data = {
+        "buses": network.buses,
+        "lines": network.lines,
+        "generators": network.generators,
+        "loads": network.loads,
+    }
+    for market_config in marketdesign:
+        market_config.param_dict["grid_data"] = grid_data
+        world.add_market(mo_id, market_config)
+
+    # naive_eom
+    default_strategy = {mc.market_id: "naive_redispatch" for mc in marketdesign}
+
+    bidding_strategies = {
+        "hard coal": default_strategy,
+        "lignite": default_strategy,
+        "oil": default_strategy,
+        "gas": default_strategy,
+        "biomass": default_strategy,
+        "nuclear": default_strategy,
+        "wind": default_strategy,
+        "solar": default_strategy,
+        "demand": default_strategy,
+    }
+
+    world.add_unit_operator("powerplant_operator")
+    for _, generator in network.generators.iterrows():
```

Comment:
we should allow the use of external csv files as extra inputs as well. would it work?

Reply:
As said in the comment above, you can add your extra CSV inputs to the pypsa network beforehand and have them run in the simulation?
Wouldn't this be enought?

Reply:
my main point is to be able to define bidding startegies for all units, without this functionality I don't see any reason why have such a loader. Network plus market is awesome combination, but markets and bidding strategies need to be configurable, if we want to do it right.

Reply:
I just addressed this in f5a2bd18393f07b4fd9fa397c147fa23b7057864
So you can now configure your market design and the used bidding strategies individually.